### PR TITLE
Recording response errors

### LINKED ISSUE #180: Recording Response Errors
In #45 the InvalidRecordingId error const was added, however this is not used or documented anywhere in the BBB repos, except the archived bbb-lti repo.

However the notFound error is missing in all recording responses.

### PR DESCRIPTION
**Pull request by SamuelWei**
Fixes #180 

- Add `isNotFound()` to PublishRecordingsResponse, UpdateRecordingsResponse and DeleteRecordingsResponse
- Deprecated `isInvalidId()` in DeleteRecordingsResponse
- Add `isNoRecordings()`, `isInvalidLang()`, `isInvalidKind()` in PutRecordingTextTrackResponse (see https://docs.bigbluebutton.org/development/api/#putrecordingtexttrack)
- Change the behaviour of json responses to non-existent json keys by no longer throwing exceptions but returning an empty string instead (same as the xml responses)

### CODE REVIEW

**Comment by SamuelWei** [comment]
Should be part of 5.x due to the deprecation

File: tests/unit/Responses/DeleteRecordingsResponseTest.php [hunk 1]
```diff
@@ -49,4 +49,14 @@ public function testDeleteRecordingsResponseTypes()
         $this->assertEachGetterValueIsString($this->delete, ['getReturnCode']);
         $this->assertEachGetterValueIsBoolean($this->delete, ['isDeleted']);
     }
+
+    public function testNotFoundError()
```
Comment (FelixJacobi) [inline tests/unit/Responses/DeleteRecordingsResponseTest.php:53]:
```suggestion
    public function testNotFoundError(): void
```

File: tests/unit/Responses/PutRecordingTextTrackResponseTest.php [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+ */
+
+namespace BigBlueButton\Parameters;
+
+use BigBlueButton\Responses\PutRecordingTextTrackResponse;
+use BigBlueButton\TestCase;
+
+final class PutRecordingTextTrackResponseTest extends TestCase
+{
+    public function testUploadSuccess()
```
Comment (FelixJacobi) [inline tests/unit/Responses/PutRecordingTextTrackResponseTest.php:27]:
```suggestion
    public function testUploadSuccess(): void
```

File: tests/unit/Responses/PutRecordingTextTrackResponseTest.php [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+              }
+            }';
+
+        $response = new PutRecordingTextTrackResponse($json);
+
+        $this->assertTrue($response->failed());
+        $this->assertTrue($response->isInvalidLang());
+    }
+
+    public function testInvalidKind()
```
Comment (FelixJacobi) [inline tests/unit/Responses/PutRecordingTextTrackResponseTest.php:111]:
```suggestion
    public function testInvalidKind(): void
```

**Review by FelixJacobi — APPROVED** [review]

**Pull request merged**